compiler/src/lnntoamm/Fn.ts

```diff
@@ -76,25 +77,25 @@ export default class Fn {
   }
 
   static fromFunctionsAst(ast: LPNode, scope: Scope): Fn {
-    scope = new Scope(scope);
+    const fnSigScope = new Scope(scope);
+
     let retTy: Type;
     if (ast.get('optreturntype').has()) {
       const name = ast.get('optreturntype').get('fulltypename');
-      retTy = Type.getFromTypename(name, scope);
+      retTy = Type.getFromTypename(name, fnSigScope, { isTyVar: true });
       if (retTy === null) {
         throw new Error(`Type not in scope: ${name.t.trim()}`);
       }
-      if (retTy.dupIfNotLocalInterface() !== null) {
-        // TODO: figure out how to prevent type erasure while allowing
-        // eg the generic identity function. Or just wait until generic
-        // fn type parameters.
-        throw new Error(`type erasure is illegal`);
-      }
+      // if (retTy.dup() !== null) {
+      //   // TODO: figure out how to prevent type erasure while allowing
+      //   // eg the generic identity function. Or just wait until generic
+      //   // fn type parameters.
+      //   throw new Error(`type erasure is illegal`);
+      // }
```

Comment:
Is this still relevant?

```suggestion
```

Reply:
Since type erasure isn't desired, it might be useful to keep a comment summarizing this block. I'll go ahead and leave a TODO for type erasure